### PR TITLE
Implemented updates for backend usage

### PR DESCRIPTION
**Pull request by b3kN**
- Updated createFootage to parse and save clips
  - TODO: Validate download format from YouTube
- Implemented Python script for testing
  - Verification required on 30 vs 60 fps
- Merged in the web branch and previous changes to master
- Implemented logic to use exported next application in production (pnpm start)
- Implemented logic to use dev systems (pnpm dev)
- Implemented basic OpenAPI documentation page for backend API

### CODE REVIEW

File: apps/backend/package.json [hunk 1]
```diff
@@ -4,10 +4,11 @@
   "description": "Back end application to process videos based on YouTube URL for AI/ML model usage.",
   "scripts": {
     "db": "docker compose up",
-    "dev": "ts-node-dev --respawn --pretty --transpile-only src/index.ts",
-    "start": "ts-node dist/index.js",
+    "dev": "NODE_ENV=dev ts-node-dev --respawn --pretty --transpile-only src/index.ts",
+    "start": "NODE_ENV=production ts-node dist/index.js",
```
Comment (Huskydog9988) [inline apps/backend/package.json:8]: just a note, this prevents windows devs from using these scripts, we should proably be providing dev containers in this case, or some other reasonable substitute.

File: apps/backend/src/services/clips.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    await fse.copy(clipDir, clipLocation);
+  }
+}
+
+export function parseClips(uuid: string, video: string): void {
+  if (!fs.existsSync('clips')) {
+    fs.mkdirSync('clips');
+  }
+
+  const pyPro = spawn('python3', ['autoClip.py', 'test-new.mp4', 'clips', '1']);
```
Comment (Huskydog9988) [inline apps/backend/src/services/clips.ts:45]: what does the python code exactly do in relation to this?

File: apps/web/additional.d.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+declare namespace NodeJS {
+    interface ProcessEnv {
+        DISCORD_CLIENT_ID: string;
+    }
+  }
+  declare namespace NodeJS {
+    interface ProcessEnv {
+        DISCORD_CLIENT_SECRET: string;
+    }
+  }
```
Comment (Huskydog9988) [inline apps/web/additional.d.ts]: am i missing something here?

File: apps/web/components/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  Flex,
+  Tag,
+} from '@chakra-ui/react';
+import { ReactNode } from 'react';
+import Image from 'next/image';
+
+const Logo = () => {
+  return (
+    <Image
+      src="/android-chrome-256x256.png"
```
Comment (Huskydog9988) [inline apps/web/components/Footer.tsx:17]: in the future, can we switch to svg? (it scales better plus smaller file size)

File: apps/web/context/GlobalContext.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  user: {
+    auth: {
+      discord: {
+        id: undefined,
+        connected: false,
+        iconURL: '',
+      },
+    },
+  },
+});
```
Comment (Huskydog9988) [inline apps/web/context/GlobalContext.tsx:23]: i assume next auth populates this?

File: apps/web/pages/_app.tsx [hunk 6]
```diff
@@ -1,6 +1,53 @@
...
+      auth: {
+        discord: {
+          id: undefined,
+          connected: false,
+          iconURL: '',
+        },
+      },
+    },
+  };
+}
```
Comment (Huskydog9988) [inline apps/web/pages/_app.tsx:33]: i assume this is gonna pull from apps/web/context/GlobalContext.tsx in the future?

File: apps/web/tsconfig.json [hunk 7]
```diff
@@ -5,7 +5,15 @@
     "**/*.ts",
     "**/*.tsx",
     ".next/types/**/*.ts"
-  ],
+, "pages/api/auth/[...nextauth].js"  ],
```
Comment (Huskydog9988) [inline apps/web/tsconfig.json]: that looks like a type error

File: package.json [hunk 8]
```diff
@@ -27,5 +30,12 @@
...
     "node": ">=14.0.0"
   },
-  "packageManager": "pnpm@7.15.0"
+  "packageManager": "pnpm@7.15.0",
+  "dependencies": {
+    "@heroicons/react": "^2.0.13",
+    "install": "^0.13.0",
+    "next-auth": "^4.16.4",
+    "node-fetch": "^3.3.0",
+    "npm": "^9.1.1"
```
Comment (Huskydog9988) [inline package.json:39]: i don't think this is right

File: packages/ui/package.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "ui",
+  "version": "0.0.0",
+  "main": "./index.tsx",
+  "types": "./index.tsx",
+  "license": "MIT",
```
Comment (Huskydog9988) [inline packages/ui/package.json:6]: wrong license

File: turbo.json [hunk 10]
```diff
@@ -12,6 +12,14 @@
...
     "clean": {
       "cache": false
+    },
+    "dev": {
+      "cache": false
+    },
+    "start": {
+      "dependsOn": ["^build"],
+      "cache": false,
+      "env": ["PORT", "HOST", "DB_URL", "FS_LOCATION", "NODE_ENV"]
```
Comment (Huskydog9988) [inline turbo.json:22]: no cache?

File: apps/backend/src/controllers/footage.controller.ts [hunk 11]
```diff
@@ -1,84 +1,61 @@
...
-        username,
         youtubeUrl: url,
-        isCsgoFootage: false,
-        isAnalyzed: false,
       };
 
-      await Footage.create(footageInput);
+      Footage.create(footageInput);
+
+      parseClips(footageId, `${footageId}.mp4`);
```
Comment (Huskydog9988) [inline apps/backend/src/controllers/footage.controller.ts:55]: comment?

**Review by Huskydog9988 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+/apps/backend/**  @b3kN
+
+# Web project:
+/apps/web/**  @tsundDEV
+
+# Desktop project:
+/apps/desktop/**  @tsundDEV
+
+# Windows forms areas:
+/windowsforms/** @adegeo @dotnet/dotnet-winforms
```
[the remark this answers is not in the document]
Comment (b3kN) [reply .github/CODEOWNERS]: Copy pasta - It has been removed

File: apps/backend/package.json (hunk 1 above)
Comment (b3kN) [reply apps/backend/package.json:8]: True - WSL ftw

File: package.json (hunk 8 above)
Comment (b3kN) [reply package.json:39]: https://github.com/waldo-vision/waldo/blob/web/package.json

Comment (Huskydog9988) [reply package.json:39]: yea, these are in `./package.json`, they should be in the `app/web/package.json` (i presume)

Comment (b3kN) [reply package.json:39]: For sure, it was merged in but I hadn't attended to any web yet. I can fix this up quick

File: apps/web/additional.d.ts (hunk 3 above)
Comment (b3kN) [reply apps/web/additional.d.ts]: I'll clean this up

Comment (Huskydog9988) [reply apps/web/additional.d.ts]: i assumed this had something to do with public env vars, but i guess i was wrong?

Comment (b3kN) [reply apps/web/additional.d.ts]: To be honest I'm not sure what it is being used for yet, but I figured the namespace didn't need to be declared twice

File: apps/backend/autoClip.py [hunk 13]
```diff
@@ -0,0 +1,216 @@
+"""
```
[the remark this answers is not in the document]
Comment (b3kN) [reply apps/backend/autoClip.py:1]: No doubt - I have my initial draft for the basic backend items: https://gist.github.com/b3kN/bef6d8f38a8417be7bc69d5360750e95

I need to update a few things and include the detailed expectations

File: apps/backend/src/services/clips.ts (hunk 2 above)
Comment (b3kN) [reply apps/backend/src/services/clips.ts:45]: Well test-new.mp4 was my happy path, so I'll get that corrected. I will provide some notes in the comments for this section and the python script later. 

The python script gathers clips around the killshots found in CSGO footage based on the killfeed in the view. It grabs the frames around and including the kill shot into trimmed images to use in sequence for the AI model building. The node logic then stores the files into the pre-determined storage location from the env.

Related Task: https://github.com/waldo-vision/waldo.footage.collection/issues/9

For the short term, this fulfills the initial clip creation requirements:
```
Clip collection requirements:

1. Validate CS:GO footage
2. Create clips on kill shots
3. Clips should be cropped to 256x256 around the crosshair 
```

Comment (Huskydog9988) [reply apps/backend/src/services/clips.ts]: thanks, hopefully we can get to commenting later, ik i haven't been great at it

File: packages/eslint-config-custom/package.json [hunk 14]
```diff
@@ -2,7 +2,7 @@
   "name": "eslint-config-custom",
   "version": "0.0.0",
   "main": "index.js",
-  "license": "MIT",
+  "license": "MDL",
```
Comment (Huskydog9988) [inline packages/eslint-config-custom/package.json:5]:
```suggestion
  "license": "MPL-2.0",
```

**Review by Huskydog9988 — APPROVED** [review]

**Review by Huskydog9988 — APPROVED** [review]

File: apps/web/components/Navbar.tsx [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+interface NavItem {
+  label: string;
+  href: string;
+}
+
+const NAV_ITEMS: Array<NavItem> = [
+  {
+    label: 'Upload',
+    href: '/upload',
+  },
```
Comment (tsundDEV) [inline apps/web/components/Navbar.tsx:182]: Remove Upload in navigation

File: apps/web/pages/index.tsx [hunk 16]
```diff
@@ -1,71 +1,238 @@
...
+              colorScheme={gc.user.auth.discord.connected ? 'green' : 'purple'}
+            >
+              Discord
+            </Button>
+            <Button
+              colorScheme={gc.user.auth.discord.connected ? 'purple' : 'red'}
+              disabled={!gc.user.auth.discord.connected ? true : false}
+            >
+              Submit
+            </Button>
```
Comment (tsundDEV) [inline apps/web/pages/index.tsx:228]: Add functionality to buttons

**Review by tsundDEV — CHANGES_REQUESTED** [review]
**Front end Changes**

- Remove the upload in navigation
- Add functionallity to the discord button on the modal

**Pull request merged**